security-framework/src/key.rs

```diff
@@ -24,6 +29,308 @@ impl_TCFType!(SecKey, SecKeyRef, SecKeyGetTypeID);
 unsafe impl Sync for SecKey {}
 unsafe impl Send for SecKey {}
 
+#[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+#[allow(missing_docs)]
+/// The algorithms that cryptographic keys enable.
+pub enum Algorithm {
```

Comment:
I imagine Apple will add more algorithms in the future. Can you make this enum non_exhaustive?

Reply:
Also there names are repeated a few times. Maybe you could use a macro to reduce the boilerplate?

```rust
macro_rules! names {
    ($($i:ident => $x:ident),*) => {
    	extern "C" {
       		$(static $x: u32;)*
    	}
        enum Enum {
            $( $i ),*
        }
        fn from(m: Enum) -> u32 {
            unsafe { match m {
                $( $i => $x, )*
            } }
        }
    }
}

names! {
    foo => macFoo, bar => macBar, baz => macBaz
}
```